### PR TITLE
Add @yasyf submission (Runs ~100% faster than @merykitty's locally)

### PR DESCRIPTION
**Pull request by yasyf**
```shell
$ hyperfine --warmup 1 --runs 3 "./calculate_average_yasyf.sh"
Benchmark 1: ./calculate_average_yasyf.sh
  Time (mean ± σ):      1.583 s ±  0.025 s    [User: 19.890 s, System: 1.112 s]
  Range (min … max):    1.569 s …  1.612 s    3 runs
```

@merykitty's submission:

```shell
$ hyperfine --warmup 1 --runs 3 "./calculate_average_merykitty.sh"
Benchmark 1: ./calculate_average_merykitty.sh
  Time (mean ± σ):      3.620 s ±  0.199 s    [User: 44.306 s, System: 1.311 s]
  Range (min … max):    3.498 s …  3.849 s    3 runs
```

- [x] Tests pass (`./test.sh <username>`)
- [x] All formatting changes by the build are committed
- [x] Output matches that of calculate_average_baseline.sh
* Execution time: 1.583s
* Execution time of reference implementation: 2m42.5s

### CODE REVIEW

File: src/main/java/dev/morling/onebrc/CalculateAverage_yasyf.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.TreeMap;
+
+import sun.misc.Unsafe;
+
+public class CalculateAverage_yasyf {
+    private static final int CORES = Runtime.getRuntime().availableProcessors() * 2;
+    private static final int N_CITIES = 1000;
```
Comment (Bananeweizen) [inline src/main/java/dev/morling/onebrc/CalculateAverage_yasyf.java:35]: > There is a maximum of 10,000 unique station names

Doesn't fit the rules, if that constant is the limit of unique cities.

**Comment by sharpobject** [comment]
sum can overflow on legal inputs. also, looks like this is ~100% faster than that other submission.

File: src/main/java/dev/morling/onebrc/CalculateAverage_yasyf.java (hunk 1 above)
Comment (yasyf) [reply src/main/java/dev/morling/onebrc/CalculateAverage_yasyf.java]: fixed! thx

**Comment by yasyf** [comment]
@sharpobject fixed, thx for that. @gunnarmorling lmk what you think

**Comment by yasyf** [comment]
@gunnarmorling ran it on a test-like box, results are less dramatic than local,  but still not bad:

```shell
[build@control 1brc]$ hyperfine --warmup 1 --runs 3 "./calculate_average_yasyf.sh"
Benchmark 1: ./calculate_average_yasyf.sh
  Time (mean ± σ):      6.493 s ±  0.126 s    [User: 43.337 s, System: 1.464 s]
  Range (min … max):    6.348 s …  6.580 s    3 runs
```

**Comment by yasyf** [comment]
@gunnarmorling [vectorizing](https://github.com/gunnarmorling/1brc/pull/204/commits/f9d4446234951080ba91a3fa4ef0b4623197f972) helped... a lot... can you confirm these results? running this on a similar instance.

```shell
[build@control 1brc]$ hyperfine --warmup 1 --runs 3 "./calculate_average_yasyf.sh"
Benchmark 1: ./calculate_average_yasyf.sh
  Time (mean ± σ):      1.280 s ±  0.017 s    [User: 5.856 s, System: 0.256 s]
  Range (min … max):    1.261 s …  1.294 s    3 runs
```

**Comment by gunnarmorling** [comment]
Hey, could you please rebase to current main, squash everything into one commit and run ./tests.sh yasyf? There's still a difference reported in the output due to JVMCI.

**Comment by gunnarmorling** [comment]
Gentle ping on this one.

**Comment by yasyf** [comment]
@gunnarmorling sorry been a crazy week! will come back to this one tomorrow :)

**Comment by gunnarmorling** [comment]
Hey @yasyf, no worries. I'm gonna close this one for now as I haven't heard back and I need to clean the slate a bit. Please feel free to reopen when you have the time to rebase it and make sure it passes all the tests. Thx!

**Pull request closed without merging**